app/main.py

```diff
@@ -1,25 +1,82 @@
-from abc import ABC
+from abc import ABC, abstractmethod
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __get__(self, instance: int, owner: object) -> None:
+        return getattr(instance, self._name)
+
+    def __set__(self, instance: object, value: int) -> None:
+        if not isinstance(value, int):
+            raise ValueError("Value must be an integer")
+        if not self.min_amount <= value <= self.max_amount:
+            raise ValueError(f"Value must be between {self.min_amount}"
+                             f" and {self.max_amount}")
+        setattr(instance, self._name, value)
+
+    def __set_name__(self, owner: object, name: str) -> None:
+        self._name = name
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(self, age: "IntegerRange",
+                 weight: "IntegerRange",
+                 height: "IntegerRange"
+                 ) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
+
+    @abstractmethod
+    def validate(self, visitor: "Visitor") -> bool:
+        pass
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    def __init__(self) -> None:
+        super().__init__(IntegerRange(4, 14),
+                         IntegerRange(20, 50),
+                         IntegerRange(80, 120))
+
+    def validate(self, visitor: "Visitor") -> bool:
+        return all([
+            self.age.min_amount <= visitor.age <= self.age.max_amount,
+            self.weight.min_amount <= visitor.weight <= self.weight.max_amount,
+            self.height.min_amount <= visitor.height <= self.height.max_amount
+        ])
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    def __init__(self) -> None:
+        super().__init__(IntegerRange(14, 60),
+                         IntegerRange(50, 120),
+                         IntegerRange(120, 220))
+
+    def validate(self, visitor: "Visitor") -> bool:
+        return all([
+            self.age.min_amount <= visitor.age <= self.age.max_amount,
+            self.weight.min_amount <= visitor.weight <= self.weight.max_amount,
+            self.height.min_amount <= visitor.height <= self.height.max_amount
+        ])
 
 
 class Slide:
-    pass
+    def __init__(self, name: str,
+                 limitation_class: "SlideLimitationValidator"
+                 ) -> None:
+        self.name = name
+        self.limitation_validator = limitation_class()
+
+    def can_access(self, visitor: "Visitor") -> bool:
```

Comment:
Do not use classes for annotations in double quotes, omit them
```suggestion
    def can_access(self, visitor: Visitor) -> bool:
```
In case you face errors, add `from __future__ import annotations`